crates/bevy_color/src/color_gradient.rs

```diff
@@ -0,0 +1,104 @@
+use crate::Mix;
+use bevy_math::curve::{cores::EvenCoreError, Curve, Interval, SampleCurve};
+
+/// A curve whose samples are defined by a collection of colors.
+#[derive(Clone, Debug)]
+pub struct ColorCurve<T: Mix + Clone, I> {
+    curve: SampleCurve<T, I>,
+}
+
+impl<T, I> ColorCurve<T, I>
+where
+    T: Mix + Clone,
+{
+    /// Create a new [`ColorCurve`] from a collection of [mixable] types and a mixing function. The
+    /// domain of this curve will always be `[0.0, len - 1]` where `len` is the amount of mixable
+    /// objects in the collection.
+    ///
+    /// This fails if there's not at least two mixable things in the collection.
+    ///
+    /// [mixable]: `Mix`
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_color::palettes::basic::*;
+    /// # use bevy_color::Mix;
+    /// # use bevy_color::Srgba;
+    /// # use bevy_color::ColorCurve;
+    /// # use bevy_math::curve::Interval;
+    /// # use bevy_math::curve::Curve;
+    /// let broken = ColorCurve::new([RED], Srgba::mix);
+    /// assert!(broken.is_err());
+    /// let gradient = ColorCurve::new([RED, GREEN, BLUE], Srgba::mix);
+    /// assert!(gradient.is_ok());
+    /// assert_eq!(gradient.unwrap().domain(), Interval::new(0.0, 2.0).unwrap());
+    /// ```
```

Comment:
```suggestion
    /// Create a new [`ColorCurve`] from a collection of [mixable] types.
    /// The domain of this curve will always be `[0.0, len - 1]` where `len` is the amount of mixable
    /// objects in the collection.
    ///
    /// This fails if there's not at least two mixable things in the collection.
    ///
    /// [mixable]: `Mix`
    ///
    /// # Example
    ///
    /// ```
    /// # use bevy_color::palettes::basic::*;
    /// # use bevy_color::Mix;
    /// # use bevy_color::Srgba;
    /// # use bevy_color::ColorCurve;
    /// # use bevy_math::curve::Interval;
    /// # use bevy_math::curve::Curve;
    /// let broken = ColorCurve::new([RED]);
    /// assert!(broken.is_err());
    /// let gradient = ColorCurve::new([RED, GREEN, BLUE]);
    /// assert!(gradient.is_ok());
    /// assert_eq!(gradient.unwrap().domain(), Interval::new(0.0, 2.0).unwrap());
    /// ```
```

Cleans up the documentation a little too.